Gems/PhysX/Code/Source/Pipeline/MeshGroup.h

```diff
@@ -211,7 +211,10 @@ namespace PhysX
             const ConvexDecompositionParams& GetConvexDecompositionParams() const;
 
         protected:
-            static bool VersionConverter(AZ::SerializeContext& context, AZ::SerializeContext::DataElementNode& classElement);
+            // void* parameter is used for the ClassElement to avoid the need to reference
+            // the SerializeContext::ClassElement type. That would require the include
+            // of SerializeContext
+            static bool VersionConverter(AZ::SerializeContext& context, void* classElement);
```

Comment:
I'm not saying that its wrong to remove it (it may be correct) but bear in mind that the files that O3DE has in its projects is a tiny window into all the files that exist somewhere out there in user's machines.  Searching for a particular thing in the files that are part of O3DE doesn't provide full confidence that it doesn't still exist somewhere in say, a user project.  However, it can provide evidence of how old it is / ie, before it became O3DE / multiple years ago / etc.

Reply:
Doing a check on the MeshGroup VersionConveter reflection code itself, it is only for `MeshGroup` instances of [version 1 and below](https://github.com/o3de/o3de/blame/ec52f77dde0d3e7e334810a7241adf1085b8156d/Gems/PhysX/Code/Source/Pipeline/MeshGroup.cpp#L856-L914)

At initial commit to O3DE to reflect the `MeshGroup` class, it was [Version 2](https://github.com/o3de/o3de/blame/9a63716b39d6e83ac76559a5eaedce00227fa0c9/Gems/PhysX/Code/Source/Pipeline/MeshGroup.cpp#L596-L608).

Therefore the only content that could have a lower version of the `MeshGroup` type instance serialized would be within the O3DE repo, at the initial commit time of the O3DE code.

Reply:
kill it with fire then :)

Reply:
A couple of the AutomatedTetring Physics assetinfo files that came with O3DE contains version 1 data for the `MeshGroup` class: 
https://sourcegraph.com/search?q=context:global+repo:%5Egithub%5C.com/o3de/o3de%24+version%3D%221%22+type%3D%22%7B5B03C8E6&patternType=standard&sm=1&groupBy=path

There are 2 files with Version 1 data in them currently
* [Material_PerFaceMatgerialGetsMaterial/test.fbx.assetinfo](https://github.com/o3de/o3de/blob/development/AutomatedTesting/Levels/Physics/Material_PerFaceMaterialGetsCorrectMaterial/test.fbx.assetinfo#L5)
* [Collider_MultipleSurfaceSlots/test.fbx.assetinfo](https://github.com/o3de/o3de/blob/development/AutomatedTesting/Assets/Physics/Collider_MultipleSurfaceSlots/test.fbx.assetinfo#L5)

Reply:
I used the `SerializeContextTools` convert command to load those two files using ObjectStream with the Version Converter for Version 1 and saved backout using json with the command of
```shell
<build-dir>/bin/debug/SerializeContextTools.exe convert --project-path=D:/o3de/o3de/AutomatedTesting --specializations=editor --files=D:/o3de/o3de/AutomatedTesting/Levels/Physics/Material_PerFaceMaterialGetsCorrectMaterial/test.fbx.assetinfo;D:/o3de/o3de/AutomatedTesting/Assets/Physics/Collider_MultipleSurfaceSlots/test.fbx.assetinfo  -ext=json
```

Reply:
The only way for users to have left over assets that would need the version converter, is if they are trying to use fbx.assetinfo files created with Lumberyard in O3DE.
Otherwise at minimum the user would have a version 2 MeshGroup serialized ObjectStream file, where the Version Converter wouldn't do anything as it checks specify for two cases
1. That the version is less than 1 to remove the Material Rule: https://github.com/o3de/o3de/blob/development/Gems/PhysX/Code/Source/Pipeline/MeshGroup.cpp#L832-L835
2. That the version is less than 2 to move all the fields to subclasses: https://github.com/o3de/o3de/blob/development/Gems/PhysX/Code/Source/Pipeline/MeshGroup.cpp#L832-L855